src/dialog.jsx

```diff
@@ -145,13 +149,19 @@ let Dialog = React.createClass({
   componentWillReceiveProps (nextProps, nextContext) {
     let newMuiTheme = nextContext.muiTheme ? nextContext.muiTheme : this.state.muiTheme;
     this.setState({muiTheme: newMuiTheme});
+
+    if( nextProps.isOpen !== this.props.isOpen ) {
```

Comment:
I've made this to resemble the behaviour of an input fields `defaultValue` and `value`. Once the `defaultValue` is set it doesn't update when new values are being set for `defaultValue`.